File: pdu/python/nav_msgs/pdu_cdr_conv_Path.py
from .pdu_pytype_Path import Path
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..geometry_msgs.pdu_cdr_conv_Point import *
from ..geometry_msgs.pdu_cdr_conv_Pose import *
from ..geometry_msgs.pdu_cdr_conv_PoseStamped import *
from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_Path(writer: CdrWriter, src: Path):
    py_to_cdr_body_Header(writer, src.header)
    writer.write_sequence_length(src.poses)
    for elem in src.poses:
        py_to_cdr_body_PoseStamped(writer, elem)


def cdr_body_to_py_Path(reader: CdrReader, dst: Path):
    cdr_body_to_py_Header(reader, dst.header)
    dst.poses = []
    for _ in range(reader.read_uint32()):
        elem = PoseStamped()
        cdr_body_to_py_PoseStamped(reader, elem)
        dst.poses.append(elem)
    return dst


def py_to_cdr_Path(src: Path) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Path(writer, src)
    return writer.bytes()


def cdr_to_py_Path(cdr_payload) -> Path:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Path()
    return cdr_body_to_py_Path(reader, dst)

File: pdu/javascript/nav_msgs/pdu_cdr_conv_Path.js
import { Path } from './pdu_jstype_Path.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Point } from '../geometry_msgs/pdu_jstype_Point.js';
import { PduPointConverter } from '../geometry_msgs/pdu_cdr_conv_Point.js';
import { Pose } from '../geometry_msgs/pdu_jstype_Pose.js';
import { PduPoseConverter } from '../geometry_msgs/pdu_cdr_conv_Pose.js';
import { PoseStamped } from '../geometry_msgs/pdu_jstype_PoseStamped.js';
import { PduPoseStampedConverter } from '../geometry_msgs/pdu_cdr_conv_PoseStamped.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduPathConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Path } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        writer.write_sequence_length(src.poses);
        for (const elem of src.poses) {
            PduPoseStampedConverter.to_cdr_body(writer, elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Path } dst
     * @returns { Path }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        dst.poses = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.poses.push(PduPoseStampedConverter.cdr_body_to_js(reader, new PoseStamped()));
        }
        return dst;
    }

    /**
     * @param { Path } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Path }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Path());
    }
}
